app/src/main/java/com/example/cm/data/repositories/FriendRequestRepository.java

```diff
@@ -15,19 +15,45 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 import static com.example.cm.data.models.Request.RequestState.REQUEST_DECLINED;
+import static com.example.cm.data.models.Request.RequestState.REQUEST_PENDING;
 
 public class FriendRequestRepository extends Repository {
 
     private final FirebaseAuth auth = FirebaseAuth.getInstance();
     private final FirebaseFirestore firestore = FirebaseFirestore.getInstance();
     private final CollectionReference friendRequestCollection = firestore.collection(CollectionConfig.FRIEND_REQUESTS.toString());
     private final MutableLiveData<List<MutableLiveData<FriendRequest>>> mutableRequestList = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> isPending = new MutableLiveData<>();
 
     public FriendRequestRepository() {
     }
 
+    /**
+     * Get all friend requests
+     */
+    public MutableLiveData<List<MutableLiveData<FriendRequest>>> getFriendRequests() {
+        if (auth.getCurrentUser() == null) {
+            return null;
+        }
+
+        String userId = auth.getCurrentUser().getUid();
+        friendRequestCollection.whereEqualTo("receiverId", userId)
+                .orderBy("createdAt", Query.Direction.DESCENDING)
+                .addSnapshotListener(executorService, (value, error) -> {
+                    if (error != null) {
+                        return;
```

Comment:
is that the way we deal with errors ? :D
(u can ignore this here, but maybe we have to do here something later)